### ISSUE TITLE
What does the convergence of `Test metric` show?

### ISSUE THREAD

**Issue by Saransh-cpp** [lululxvi/deepxde#486]
Greetings, 

I am getting the following plot while solving a 2-dimensional PDE, and for some reason, the test metric converges very quickly -

![image](https://user-images.githubusercontent.com/74055102/149724707-66ca4fff-2cd8-4594-820e-7ace86953aca.png)

Is this a good thing? Or is my model not getting trained well?

Thank you!

**Comment by lululxvi** [comment]
Interesting. It is better to check the final result by looking at the data.

**Comment by Saransh-cpp** [comment]
Thanks, @lululxvi! These are the plots that I obtain from `"test.dat"` file. The predicted points are off by a bit but I don't understand why -

![image](https://user-images.githubusercontent.com/74055102/149980955-a995f315-25ce-4ee4-8ced-912c0f9f3845.png)

![image](https://user-images.githubusercontent.com/74055102/149981190-a6efc59b-ecb6-42f8-bb1e-d8041d9aae94.png)

The complete code -
```py
import numpy as np
import deepxde as dde
from deepxde.backend import tf
import matplotlib.pyplot as plt


k = 0.4

n = 1
m = 1
a = 1
b = 1
C = 4 / np.pi


def pde(x, u):
    u_t = dde.grad.jacobian(u, x, j=2)
    u_xx = dde.grad.hessian(u, x, i=0, j=0)
    u_yy = dde.grad.hessian(u, x, i=1, j=1)
    return u_t - k * (u_xx + u_yy)


def func(x):
    return (
        C
        * np.sin(m * np.pi * x[:, 0:1] / a)
        * np.sin(n * np.pi * x[:, 1:2] / b)
        * np.exp(-k * np.pi ** 2 * x[:, 2:3])  # (m^2/a^2  +  n^2/b^2) = 1
    )


spatial_domain = dde.geometry.Rectangle(xmin=[0, 0], xmax=[a, b])
temporal_domain = dde.geometry.TimeDomain(0, 1)
spatio_temporal_domain = dde.geometry.GeometryXTime(spatial_domain, temporal_domain)

bc = dde.DirichletBC(spatio_temporal_domain, lambda x: 0, lambda _, on_boundary: on_boundary)
ic = dde.IC(
    spatio_temporal_domain,
    lambda x: np.sin(n * np.pi * x[:, 0:1] / a),
    lambda _, on_initial: on_initial,
)


data = dde.data.TimePDE(
    spatio_temporal_domain,
    pde,
    [bc, ic],
    num_domain=2540,
    num_boundary=500,
    num_initial=500,
    num_test=2540,
    solution=func,
)

layer_size = [3] + [32] * 3 + [1]
activation = "tanh"
initializer = "Glorot uniform"
net = dde.maps.FNN(layer_size, activation, initializer)

model = dde.Model(data, net)

model.compile(
    "adam", lr=0.001, metrics=["l2 relative error"],
)
model.train(epochs=10000)
model.compile(
    "L-BFGS", metrics=["l2 relative error"],
)
losshistory, train_state = model.train()

dde.saveplot(
    losshistory,
    train_state,
    issave=True,
    isplot=True,
    test_fname="../dat_data/heat_2D.dat",
)
```

**Comment by lululxvi** [comment]
The number of training points seems small

**Comment by Saransh-cpp** [comment]
Thank you, @lululxvi, for the help! I tried increasing the training point but the problem persisted. After going through the problem again and again I realized that the ICs were wrong, which made the reference solution incorrect. Fixing them worked!
